File: ClientUI/src/app/_models/userInputParams.ts
import { User } from "./user";

export class UserInputParams{
    gender:string;
    minAge=28;
    maxAge=120;
    pageNumber=1;
    pageSize=5;
    orderBy='lastActive'
    constructor(user:User){
        this.gender=user.gender==="female" ? "male":"female";
    }
}

File: Helpers/UserInputParams.cs
namespace DaringAPI.Helpers
{
    public class UserInputParams :PaginationParams
    {
       public string CurrentUserName { get; set; }
       public string Gender { get; set; }
       public int MinAge { get; set; }=18;
       public int MaxAge { get; set; }=130;
       public string OrderBy { get; set; }="lastActive";
    }
}      